src/utils/Api.js

```diff
@@ -1,11 +1,94 @@
 export default class Api {
   constructor(options) {
-    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _defaultPromise(promise) {
+    return promise.then(this._getResult).catch(console.error);
```

Comment:
`.catch(console.error);` is not needed here

You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js` after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.